openapi.yaml

```diff
@@ -0,0 +1,319 @@
+    delete:
+      tags:
+        - todo
+      summary: Delete a todo item
+      description: Delete a todo item by ID
+      operationId: deleteTodo
+      parameters:
+        - name: id
+          in: path
+          description: ID of todo to delete
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '204':
+          description: Successfully deleted todo
+        '404':
+          description: Todo of given ID does not exist
+  /todo/{id}/complete:
+    patch:
+      tags:
+        - todo
+      summary: Complete todo
+      description: Complete todo by ID
+      operationId: completeTodo
+      parameters:
+        - name: id
+          in: path
+          description: ID of todo to complete
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '204':
+          description: Successfully completed todo
+        '404':
+          description: Todo of given ID does not exist
+  /user/register:
+    post:
+      tags:
+        - user
+      summary: Create user
+      description: Register a user account
+      operationId: createUser
+      requestBody:
+        $ref: '#/components/requestBodies/User'
+      responses:
+        '201':
+          description: Successfully created user account
+  /user/login:
+    get:
+      tags:
+        - user
+      summary: Login user
+      description: Logs user into the system
+      operationId: loginUser
+      requestBody:
+        $ref: '#/components/requestBodies/User'
+      responses:
+        '200':
+          description: Successfully logged in
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AuthResponse'
+        '404':
+          description: Account with the given email and password does not exist
+  /user/logout:
+    get:
+      tags:
+        - user
+      summary: Log out user
+      description: Logs out the currently logged in user
+      operationId: logoutUser
+      responses:
+        '200':
+          description: Successfully logged out user
+  /user/{email}:
+    put:
+      tags:
+        - user
+      summary: Update account email
+      description: Update the email linked to the currently logged in account
+      operationId: updateUser
+      parameters:
+        - name: email
+          in: path
+          description: The email to be updated
+          required: true
+          schema:
+            type: string
+      responses:
+        '204':
+          description: Successfully updated account email
+  /user/delete:
+    delete:
+      tags:
+        - user
+      summary: Delete user
+      description: Delete the currently logged in user account
+      operationId: deleteUser
+      responses:
+        '204':
+          description: Successfully deleted user account
+components:
+  schemas:
+    TodoRequest:
+      required:
+        - title
```

Comment:
Id the title is required in the create todo operation , then get operation also should guarantee that information is always available , data consistency is key APi design consideration